DelvEdit/src/com/interrupt/dungeoneer/editor/EditorApplication.java

```diff
@@ -433,18 +434,63 @@ public void loadMonsters() {
 		}
 	}
 
+	/** Creates an empty level with given `width` and `height`. */
 	public void createEmptyLevel(int width, int height) {
 		level = new Level(width, height);
+
+		Tile t = new Tile();
+		t.floorHeight = -0.5f;
+		t.ceilHeight = 0.5f;
+		level.setTile(width / 2, height / 2, t);
+
+		cleanEditorState();
+	}
+
+	/** Generates a level based on a `template` level. */
+	public void generateLevelFromTemplate(Level template) {
+		level.editorMarkers.clear();
+		level.entities.clear();
+		level.non_collidable_entities.clear();
+		level.static_entities.clear();
```

Comment:
Feels like this should be a method? Potentially `Level.clear()`

Reply:
Yes, makes sense.